### PR TITLE
DOC: Improve docs for `jnp.angle` & `jnp.flip`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461.

### CODE REVIEW

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -716,8 +716,64 @@ def rot90(m: ArrayLike, k: int = 1, axes: tuple[int, int] = (0, 1)) -> Array:
...
-@util.implements(np.flip, lax_description=_ARRAY_VIEW_DOC)
 def flip(m: ArrayLike, axis: int | Sequence[int] | None = None) -> Array:
+  """Reverse the order of elements of an array along the given axis.
+
+  JAX implementation of :func:`numpy.flip`.
+
+  Args:
+    m: Array.
+    axis: integer or sequence of integers. Specifies along which axis or axes
+    should the array elements be reversed.
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:727]: Line 727 needs to be indented two spaces. Also, we should mention that `axis` defaults to `None`, which flips along all axes.

Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py]: Done

**Review by jakevdp — APPROVED** [review]

**Review by jakevdp — CHANGES_REQUESTED** [review]
Looks like there are some lint issues (trailing whitespace) and some doctest issues (undefined variables). Please take a look!

**Comment by rajasekharporeddy** [comment]
Fixed lint_and_typechecking errors and squashed all into single commit.
